src/Microsoft.AspNet.Mvc.ModelBinding/Binders/ArrayModelBinder.cs

```diff
@@ -19,9 +26,42 @@ public override Task<ModelBindingResult> BindModelAsync(ModelBindingContext bind
             return base.BindModelAsync(bindingContext);
         }
 
+        /// <inheritdoc />
         protected override object GetModel(IEnumerable<TElement> newCollection)
         {
+            if (newCollection == null)
+            {
+                return null;
+            }
+
             return newCollection.ToArray();
         }
+
+        /// <inheritdoc />
+        protected override void CopyToModel([NotNull] object target, IEnumerable<TElement> sourceCollection)
+        {
+            TElement[] targetArray = target as TElement[];
+            Debug.Assert(targetArray != null); // This binder is instantiated only for array model types.
+
+            if (sourceCollection != null && targetArray != null)
+            {
+                int maxIndex = targetArray.Length - 1;
+                int index = 0;
+                foreach (var element in sourceCollection)
+                {
+                    if (index > maxIndex)
+                    {
+                        break;
```

Comment:
MVC 5 seems to do nothing with arrays passed into `TryUpdateModel`.  may have similar issues with dictionaries as well.  but MVC 5 also doesn't have the equivalent of `ArrayModelBinder` and so can't avoid the problem as easily.

in any case, providing as many elements as we can is an improvement over the silent failures in MVC 5.  an error would be incompatible.